tests/functional/models/test_environment_variables.py

```diff
@@ -0,0 +1,235 @@
+import unittest
+
+from tests.helper import TestHelper
+
+
+class TestAppEnvironmentVariables(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.helper = TestHelper()
+        cls.balena = cls.helper.balena
+        # Wipe all apps before the tests run.
+        cls.helper.wipe_application()
+        cls.app = cls.balena.models.application.create("FooBar", "raspberry-pi2", cls.helper.default_organization["id"])
+        cls.__app_fetch_resources = ["id", "slug", "uuid"]
+        cls.app_env_var = cls.balena.models.application.env_var
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.helper.wipe_organization()
+
+    def test_01_get_emtpy_application_env_var(self):
+        self.assertIsNone(self.app_env_var.get(self.app["id"], "DOES_NOT_EXIST"))
+
+        with self.assertRaises(self.helper.balena_exceptions.ApplicationNotFound):
+            self.assertIsNone(self.app_env_var.get("org-does-not/exist", "DOES_NOT_EXIST"))
+
+    def test_02_can_create_and_retrieve_application_env_var(self):
+        for resource in self.__app_fetch_resources:
+            self.app_env_var.set(self.app[resource], f"EDITOR_BY_{resource}", f"VIM{resource}")
+            self.assertEqual(
+                self.app_env_var.get(self.app[resource], f"EDITOR_BY_{resource}"),
+                f"VIM{resource}",
+            )
+
+    def test_03_can_get_all_app_vars(self):
+        for resource in self.__app_fetch_resources:
+            app_vars = self.app_env_var.get_all_by_application(self.app[resource])
+            expected_vars = [
+                {"name": f"EDITOR_BY_{resource}", "value": f"VIM{resource}"} for resource in self.__app_fetch_resources
+            ]
+            app_vars_wo_id = [{"name": entry["name"], "value": entry["value"]} for entry in app_vars]
+
+            for var in expected_vars:
+                self.assertIn(var, app_vars_wo_id)
+
+    def test_04_can_update_app_vars(self):
+        for resource in self.__app_fetch_resources:
+            self.app_env_var.set(
+                self.app[resource],
+                f"EDITOR_BY_{resource}",
+                f"VIM{resource}_edit",
+            )
+            self.assertEqual(
+                self.app_env_var.get(self.app[resource], f"EDITOR_BY_{resource}"),
+                f"VIM{resource}_edit",
+            )
+
+    def test_05_can_remove_application_env_vars(self):
+        for resource in self.__app_fetch_resources:
+            self.app_env_var.remove(self.app[resource], f"EDITOR_BY_{resource}")
+            self.assertIsNone(
+                self.app_env_var.get(self.app[resource], f"EDITOR_BY_{resource}"),
+            )
+
```

Comment:
For completeness you may also want to add tests for `TestApplicationServiceEnvironmentVariables`

Reply:
hey @fisehara thanks for the heads up, this one was already in my list (this repo has overall less test coverage than the node-sdk one, I am still in the process of increasing it)

Reply:
done